hello-microservices-middle/src/main/java/com/matera/hellomicroservices/service/PersonService.java

```diff
@@ -0,0 +1,62 @@
+package com.matera.hellomicroservices.service;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.UUID;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.inject.Inject;
+import com.matera.hellomicroservices.converter.PersonConverter;
+import com.matera.hellomicroservices.entities.Address;
+import com.matera.hellomicroservices.entities.Person;
+import com.matera.hellomicroservices.store.PersonStore;
+
+import matera.com.hellomicroservices.core.requests.CreatePersonRequest;
+import matera.com.hellomicroservices.core.responses.CreatePersonResponse;
+import matera.com.hellomicroservices.core.responses.PersonResource;
+
+public class PersonService {
+	
+	private final PersonStore store;
+
+	@Inject
+	public PersonService(PersonStore store) {
+		this.store = store;
+	}
+	
+	public CreatePersonResponse insert(CreatePersonRequest request) {
+		
+		Person person = PersonConverter.convertToPerson(request);
+		Address address = person.getAddress();
+		
+		checkArgument(StringUtils.isNotBlank(person.getFirstName()), "the first name must not be null");
+		checkArgument(StringUtils.isNotBlank(person.getLastName()), "the last name must not be null");
+		checkArgument(StringUtils.isNotBlank(person.getEmail()), "the email must not be null");
+		checkArgument(StringUtils.isNotBlank(person.getNickName()), "the nickname must not be null");
+		checkArgument(StringUtils.isNotBlank(address.getCity()), "the city must not be null");
+		checkArgument(StringUtils.isNotBlank(address.getState()), "the state must not be null");
+		checkArgument(StringUtils.isNotBlank(address.getCountry()), "the country must not be null");		
+		checkArgument(StringUtils.isNotBlank(address.getZipCode()), "the zip code must not be null");
+		
+		Person inserted = store.insert(person);
```

Comment:
what happens in case of error? lets handle it and put the proper message on response object